src/transformers/sinon.ts

```diff
@@ -698,15 +699,52 @@ function transformMockTimers(j, ast) {
     })
 }
 
+// let stub: sinon.SinonStub -> let stub: jest.Mock
+// let spy: sinon.SinonSpy -> let spy: jest.SpyInstance
+function transformTypes(j, ast, parser) {
+  if (!isTypescript(parser)) return
+
+  ast
+    .find(j.TSTypeReference, {
+      typeName: {
+        left: {
+          name: 'sinon',
+        },
+        right: {
+          name: 'SinonStub',
+        },
+      },
+    })
+    .forEach((np) => {
+      np.node.typeName.left.name = 'jest'
+      np.node.typeName.right.name = 'Mock'
+    })
+
+  ast
+    .find(j.TSTypeReference, {
+      typeName: {
+        left: {
+          name: 'sinon',
+        },
+        right: {
+          name: 'SinonSpy',
+        },
+      },
+    })
+    .forEach((np) => {
+      np.node.typeName.left.name = 'jest'
+      np.node.typeName.right.name = 'SpyInstance'
+    })
+}
+
 export default function transformer(fileInfo: FileInfo, api: API, options) {
-  const j = api.jscodeshift
+  const j = api.jscodeshift.withParser(options.parser)
```

Comment:
I've been looking for this forever.

Reply:
Lazy web: what is this actually doing? The right parser is already selected by the CLI, so I’m curious what this is doing. :)

Reply:
> Lazy web: what is this actually doing?

this took a while for me to figure out as well.  this PR adds the first test case in which the code fed to `expectTransformation()` is not parseable by the `babel` parser.  and it turns out that passing `{ parser: 'ts' }` to tests aren't actually causing the `ts` parser to be used instead of `babel`.  all tests were still using `babel`.

> The right parser is already selected by the CLI, so I’m curious what this is doing. :)

without this, typescript in the input to `expectTransformation()` yields the cryptic:

```
  ● updates types › rewrites types for mocks & stubs

    SyntaxError: Missing semicolon. (4:15)

      at Object._raise (node_modules/@babel/parser/src/parser/error.js:147:45)
```

with this line the test succeeds because the correct parser is selected.  I just tried moving this to `src/utils/test-helpers.ts` instead of in the "prod" code but it isn't working.

I can keep trying to find a way to make this work in test-only code or we can land for now and I'll continue as a followup.

Reply:
ok, moved this `withTransform()` call to test-only code

Reply:
👏